if all iterations fail | improve console output

zacharopoulou opened the pull request:
### **For iterations:**

Look at run.sh script for an extra "filtering" step

****From Feedback**:**

**(1)**     The following warnings are cluttering the output terminal when creating the container, or at least at the beginning of the run:

/opt/conda/envs/agentomics-env/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py:2249: UnsupportedFieldAttributeWarning: The 'repr' attribute with value False was provided to the `Field()` function, which has no effect in the context it was used. 'repr' is field-specific metadata, and can only be attached to a model field using `Annotated` metadata or by assignment. This may have happened because an `Annotated` type alias using the `type` statement was used, or if the `Field()` function was attached to a single member of a union type.
  warnings.warn(
/opt/conda/envs/agentomics-env/lib/python3.10/site-packages/pydantic/_internal/_generate_schema.py:2249: UnsupportedFieldAttributeWarning: The 'frozen' attribute with value True was provided to the `Field()` function, which has no effect in the context it was used. 'frozen' is field-specific metadata, and can only be attached to a model field using `Annotated` metadata or by assignment. This may have happened because an `Annotated` type alias using the `type` statement was used, or if the `Field()` function was attached to a single member of a union type.
  warnings.warn(

**Done.** Gave -e PYTHONWARNINGS=ignore to docker runs at run.sh script

**(2)** While building the image, not seeing any output for more than 5 minutes cause a feeling of the program being stuck. (This is caused by the --progress=quiet)

**Done**. Simply removed the argument

(3) In the agent messages on the terminal, the Step prompt should have a different color, right now it seems part of the output of the previous step. 

**Done**. Simply included some messages with the Steps.

(4) W&B log when doing test eval is showing a warning (if W&B not set up) saying that the run cannot be resumed. Should be silent.

Nothing is done here

(5) In the interactive run, it's not clear what we are asking for the validation metric. We are not giving any hint regarding what each metric means/how it is calculated. In general, it needs some more explanation why we are asking for this parameter and how it will be used for.

Must talk also about that

**My own suggestions:**

**(1)** Maybe tell to the user that is preferable to run the pipeline with at least 5 iterations or for at least 2 hours?

**Done**

**### For model filtering:**

I excluded some models, but most important step was to clarify that we don't want in the list models of thinking or research.

zacharopoulou commented on src/utils/providers/openrouter_provider.py:
```diff
@@ -60,31 +60,41 @@ def get_filtered_models(self, limit: int = 20) -> Optional[List[Dict]]:
+            description = model.get("description", "").lower()
+            research_keywords = ["deep research", "research model"]
+            if any(keyword in description for keyword in research_keywords) or \
+                   "deepresearch" in model_id:
```
good point. yes

dimostzim commented on src/utils/providers/openrouter_provider.py:
```diff
@@ -60,31 +60,41 @@ def get_filtered_models(self, limit: int = 20) -> Optional[List[Dict]]:
+            "ibm-granite/granite-4.0-h-micro",
+            "baidu/ernie-4.5-21b-a3b",
+            "perplexity/sonar-pro-search",
+                           ]
```
no its fine that is why we have this list so we can exclude things that do not work, no need to automatically exclude models

dimostzim commented on src/run_logging/wandb_setup.py:
```diff
@@ -35,6 +56,12 @@ def setup_logging(config, dir=None):
 
 def resume_wandb_run(config, dir=None):
+
+    if not wandb_is_configured():
```
Simply check if config.wandb_run_id exists first. if it doesn't exist, wandb wasn't used during agent run, so just return None immediately.

dimostzim commented on src/run_logging/wandb_setup.py:
```diff
@@ -35,6 +56,12 @@ def setup_logging(config, dir=None):
         return False
 
 def resume_wandb_run(config, dir=None):
```
check config.wandb_run_id exists, then env vars, then login. All
  failures return None silently.

dimostzim commented on src/run_logging/wandb_setup.py:
```diff
@@ -2,11 +2,32 @@
+        return True
+
+    # netrc present → usually means wandb login was done
+    netrc_path = Path.home() / ".netrc"
```
.netrc could contain credentials for other services, not just wandb

zacharopoulou commented:
Added dotenv.load_dotenv() in resume_wandb_run to ensure env vars are available in the test-eval process.

Added a guard to skip resuming wandb silently unless WANDB_API_KEY, WANDB_PROJECT_NAME, and WANDB_ENTITY are all set.

Added a guard to skip resume when config.wandb_run_id is missing.

Switched resume="must" to "allow" so test evaluation doesn’t send out warnings/errors when a run cannot be resumed.

dimostzim commented on src/run_logging/wandb_setup.py:
```diff
@@ -56,26 +36,25 @@ def setup_logging(config, dir=None):
-        print("W&B login failed - cannot resume logging run")
         return False
     else:
         wandb.init(
```
Wrap wandb.init() in try/except CommError (like setup_logging) and return None on exceptions. Right now a CommError can still crash test evaluation
